<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8" />
<title>Selection in Genes</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<!-- Favicon --> 
<!--link rel="shortcut icon" href="//images/favicon.ico"--> 

<!-- JQuery and Bootstrap JS (necessary for collapsing navbar) --> 
<script src="/Selection-in-Genes-Course/js/jquery-3.6.0.min.js"></script> 
<script src="/Selection-in-Genes-Course/bootstrap/js/bootstrap.bundle.min.js"></script> 

<!-- Bootstrap Icons -->
<link rel="stylesheet" type="text/css" href="/Selection-in-Genes-Course/bootstrap/icons/bootstrap-icons.css">

<!-- KaTeX -->
<link rel="stylesheet" href="/Selection-in-Genes-Course/css/katex.css">
<link href="css/style.css" rel="stylesheet" type="text/css">
<script src="/Selection-in-Genes-Course/js/katex.min.js"></script>
</head>

<body>
<div class="container">
  <div class="row">
    <div class="col-md-11">
      <div class="bigtitle logobox"> Detecting Selection in Genes </div>
    </div>
  </div>
  <div class="bigspacer"></div>
  <div class="row">
    <div>
      <h1 id="lecture-slides">Lecture slides</h1>
      Below are the lecture slides for the course "Selection in Genes" that ran in 2022 in Lund, Sweden, by Prof Dr Maria Anisimova and Dr Jūlija Pečerska from ZHAW, Zürich, Switzerland.
      <div class="smallspacer"></div>
      <ul>
        <li><a
              href="https://github.com/acg-team/Selection-in-Genes-Course/raw/gh-pages/lectures/01%20Modeling%20Molecular%20Evolution%202022opt.pdf">Markov
          Models of Sequence Evolution</a></li>
        <li><a
              href="https://github.com/acg-team/Selection-in-Genes-Course/raw/gh-pages/lectures/02%20Hypothesis%20Testing%202022%20opt.pdf">Hypothesis Testing and Model Selection</a></li>
        <li><a
              href="https://github.com/acg-team/Selection-in-Genes-Course/raw/gh-pages/lectures/03%20Phylogeny%20Inference%202022%20opt.pdf">Phylogenetic
          Inference</a></li>
        <li> <a
              href="https://github.com/acg-team/Selection-in-Genes-Course/raw/gh-pages/lectures/04%20PositiveSelection%202022%20opt.pdf">Natural
          Selection and Codon Models</a> </li>
        <li> <a
              href="https://github.com/acg-team/Selection-in-Genes-Course/raw/gh-pages/lectures/05%20DetectSelection%20BranchModels%202022%20opt.pdf">Modelling
          Variability of Selection</a></li>
        <li> <a
              href="https://github.com/acg-team/Selection-in-Genes-Course/raw/gh-pages/lectures/06%20DetectSelection%20SiteModels%202022%20opt.pdf">Site
          Models, Detecting Residues under Selection</a></li>
        <li> <a
              href="https://github.com/acg-team/Selection-in-Genes-Course/blob/gh-pages/lectures/07%20DetectSelection%20BrachSiteModels%202022%20opt.pdf">Detecting
          Episodic Positive Selection</a></li>
        <li> <a
              href="https://github.com/acg-team/Selection-in-Genes-Course/raw/gh-pages/lectures/08%20Selection_Pipelines%202022%20opt.pdf">Pipelining
          for Large-Scale Analyses</a></li>
      </ul>
    </div>
    <div class="bigspacer"></div>
    <div>
      <h1 id="tutorials">Tutorials</h1>
      <ul>
        <li><a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/parameter-estimation/">Parameter
          Estimation in PhyML</a></li>
        <li><a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/model-selection/">Model
          Selection in PhyML</a></li>
        <li><a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/tree-search/">ML
          Tree Inference in
          PhyML</a></li>
        <li><a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/simple-codon-model/">One Ratio
          Model Inference in PAML</a></li>
        <li><a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/branch-model/">Branch
          Models in PAML</a></li>
        <li><a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/site-model/">Site Models
          in PAML</a></li>
        <li><a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/branch-site-model/">Branch-site
          Model Inference in PAML</a></li>
        <li> <a href="https://acg-team.github.io/Selection-in-Genes-Course/tutorials/pipelines/">Pipelining</a> </li>
      </ul>
    </div>
    <div class="bigspacer"></div>
    <div>
      <h1 id="software">Software</h1>
      <p>The software below will be used for the exercises. </p>
      <ul>
        <li><strong> PhyML:</strong> PhyML is a software package that uses modern statistical approaches to analyse
          alignments of nucleotide or amino acid sequences in a phylogenetic framework <a
              href="http://www.atgc-montpellier.fr/phyml/">(download)</a>.</li>
        <li><strong> PAML:</strong> PAML is a package of programs for phylogenetic analyses of DNA or protein sequences
          using maximum likelihood <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">(download)</a>. </li>
        <li><strong> FigTree:</strong> FigTree is a graphical viewer of phylogenetic trees <a
              href="http://beast.community/figtree">(download)</a>.</li>
        <li><strong>AliView:</strong> AliView is an alignment viewer and editor <a
              href="https://ormbunkar.se/aliview/">(download)</a>.</li>
      </ul>
    </div>
  </div>
</div>
<div class="footerspacer"></div>
<div class="footer">
  <div class="container">
    <div class="row">
      <div class="col-md-6">
        <ul class="list-group list-group-horizontal">
          <li class="footernav list-group-item"><a class="foot"
                href="/Selection-in-Genes-Course/about/index.html">about</a></li>
          <li class="footernav list-group-item"><a class="foot"
                href="/Selection-in-Genes-Course/license/index.html">license</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>
</body>
</html>